### ISSUE TITLE
Error processing dependencies

### ISSUE THREAD

**Issue by expertsnipo** [alekdavis/PlexBackup#54]
Running the script after getting the execution policy set yields no results - no log file or anything. I can see it trying to load dependencies then it just exits. So I manually ran the Install-Module for each dependency. Now I get a little further. Here's the output I'm currently getting. I'm using PowerShell v5 not v4.

PS C:\Users\blah\PlexBackup> .\PlexBackup.ps1 -ConfigFile .\PlexBackup.ps1.json -Verbose
VERBOSE: Microsoft PowerShell v5.1.19041.1320
VERBOSE: Processing module 'ScriptVersion'.
VERBOSE: Required module name: 'ScriptVersion'.
VERBOSE: Required module 'ScriptVersion' is not loaded.
VERBOSE: Installing module 'ScriptVersion'.
VERBOSE: Importing module 'ScriptVersion'.
Error processing dependencies. Cannot load module 'ScriptVersion'. The specified module 'ScriptVersion' was not loaded because no valid module file was found in any module directory.
VERBOSE: SingleInstance module is not loaded.
VERBOSE: StreamLogging|1.2.1 module is not loaded.

**Comment by alekdavis** [comment]
Hmmm... PowerShell v5 is fine (v4 is the minimal version), but why would it install a module but not be able to import it? Makes no sense. Also there should be more entries in the log before and between the last two lines.

What happens if you run these from PowerShell?

```
Install-Module -Name 'ScriptVersion' -Force -Scope CurrentUser
Import-Module 'ScriptVersion' -Force
Install-Module -Name 'SingleInstance ' -Force -Scope CurrentUser
Import-Module 'SingleInstance ' -Force
Install-Module -Name 'StreamLogging' -Force -Scope CurrentUser -MinimumVersion '1.2.1'
Import-Module 'StreamLogging' -Force -MinimumVersion '1.2.1'
Install-Module -Name 'SingleInstance' -Force -Scope CurrentUser
Import-Module 'SingleInstance' -Force
```

**Comment by expertsnipo** [comment]
Here's my output now:
```
PS C:\Users\blah\PlexBackup> Install-Module -Name 'ScriptVersion' -Force -Scope CurrentUser
PS C:\Users\blah\PlexBackup> Import-Module 'ScriptVersion' -Force                                                       Import-Module : The specified module 'ScriptVersion' was not loaded because no valid module file was found in any       module directory.                                                                                                       At line:1 char:1                                                                                                        + Import-Module 'ScriptVersion' -Force
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (ScriptVersion:String) [Import-Module], FileNotFoundException
    + FullyQualifiedErrorId : Modules_ModuleNotFound,Microsoft.PowerShell.Commands.ImportModuleCommand
 
PS C:\Users\blah\PlexBackup> Install-Module -Name 'SingleInstance ' -Force -Scope CurrentUser
PackageManagement\Install-Package : No match was found for the specified search criteria and module name
'SingleInstance '. Try Get-PSRepository to see all available registered module repositories.
At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\1.0.0.1\PSModule.psm1:1809 char:21
+ ...          $null = PackageManagement\Install-Package @PSBoundParameters
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Microsoft.Power....InstallPackage:InstallPackage) [Install-Package], Ex
   ception
    + FullyQualifiedErrorId : NoMatchFoundForCriteria,Microsoft.PowerShell.PackageManagement.Cmdlets.InstallPackage
 
PS C:\Users\blah\PlexBackup> Import-Module 'SingleInstance ' -Force
Import-Module : The specified module 'SingleInstance ' was not loaded because no valid module file was found in any
module directory.
At line:1 char:1
+ Import-Module 'SingleInstance ' -Force
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (SingleInstance :String) [Import-Module], FileNotFoundException
    + FullyQualifiedErrorId : Modules_ModuleNotFound,Microsoft.PowerShell.Commands.ImportModuleCommand
 
PS C:\Users\blah\PlexBackup> Install-Module -Name 'StreamLogging' -Force -Scope CurrentUser -MinimumVersion '1.2.1'
PS C:\Users\blah\PlexBackup> Import-Module 'StreamLogging' -Force -MinimumVersion '1.2.1'
PS C:\Users\blah\PlexBackup> Install-Module -Name 'SingleInstance' -Force -Scope CurrentUser
PS C:\Users\blah\PlexBackup> Import-Module 'SingleInstance' -Force
```

Running `PlexBackup.ps1` again yields the same output as before.

**Comment by alekdavis** [comment]
Yeah, until you resolve the module loading issue, PlexBackup will not work. There is a workaround: download and copy module files to a local folder and pass your custom module folder path via the `-MdulePath` parameter, but I am just puzzled by the behavior of your system:  it seems to install and load the two other modules fine, just struggles with the first two. And it does not give an error on installation, just on import. Makes no sense whatsoever.

Could you try running these (one at a time) and post the output:

```
$env:PSModulePath
Get-PSRepository
Get-Module -ListAvailable
```

**Comment by expertsnipo** [comment]
OK running `$env:PSModulePath` revealed that my modules path for my current user's documents folder was not present. This is what I got:
```
PS C:\Users\blah\PlexBackup> $env:PSModulePath
;C:\Users\blah\AppData\Local\Google\Cloud SDK\google-cloud-sdk\platform\PowerShell;C:\Program Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules\;C:\Users\blah\PlexBackup\Modules;
```
I checked on my other Windows system and it had `c:\users\blah\Documents\WindowsPowerShell\Modules` in that list.
After temporarily adjusting the env var to have the path to my documents folder which contains the modules, I reran the script and it works now. I'm not sure exactly what happened but I suspect it's because I moved my user profile's library folders on this system to a different drive letter location (d:\users\blah\documents) and somehow powershell doesn't track this properly when it creates the env var. I will have to look at a permanent fix by editing the registry key to add my actual documents folder to `$env:PSModulePath`. Microsoft has this guidance:
```
$key = (Get-Item 'HKCU:\SYSTEM\CurrentControlSet\Control\Session Manager').OpenSubKey('Environment', $true)
$path = $key.GetValue('PSModulePath','','DoNotExpandEnvironmentNames')
$path += ';C:\Program Files\Fabrikam\Modules' # or '%ProgramFiles%\Fabrikam\Modules'
$key.SetValue('PSModulePath',$path,[Microsoft.Win32.RegistryValueKind]::ExpandString))
```
However that key does not exist on my system. I will keep looking, but this is definitely an issue specific to my environment.

Thanks for your help.
